alws/test_scheduler.py

```diff
@@ -22,6 +22,49 @@ def __init__(self, term_event: threading.Event,
         self._graceful_event = graceful_event
         self._alts_client = AltsClient(settings.alts_host, settings.alts_token)
 
+
+    @staticmethod
+    def __get_repos_for_test_task(task: models.TestTask):
+        repos = []
+        # Build task repos
+        build_repositories = [
+            {'name': item.name, 'baseurl': item.url}
+            for item in task.build_task.build.repos
+            if item.type == 'rpm'
+            and item.arch == task.env_arch
+        ]
+        repos.extend(build_repositories)
+
+        # Linked build repos
+        linked_build_repos = [
+            {'name': item.name, 'baseurl': item.url}
+            for build in task.build_task.build.linked_builds
+            for item in build.repos
+            if item.type == 'rpm'
+            and item.arch == task.env_arch
+        ]
+        repos.extend(linked_build_repos)
+
+        # Flavor repos
+        platform = task.build_task.platform
+        flavor_repos = [
+            {
+                'name': item.name,
+                'baseurl': item.url.replace(
+                    '$releasever',
+                    platform.distr_version
+                )
+            }
+            for flavor in task.build_task.build.platform_flavors
+            for item in flavor.repos
+            if item.type == 'rpm'
+            and item.arch == task.env_arch
+        ]
+        repos.extend(flavor_repos)
```

Comment:
can it be better, if we will iterate by new arrays and add them in `repos`?
like this
```python3
build_repositories = [...]
linked_build_repos = [...]
flavor_repos = [...]
for repo_arr in (
    build_repositories,
    linked_build_repos,
    flavor_repos,
):
    repos.extend(repo_arr)
```
it's up to you